r/src/compute.cpp

```diff
@@ -233,6 +233,12 @@ std::shared_ptr<arrow::compute::FunctionOptions> make_compute_options(
                                      max_replacements);
   }
 
+  if (func_name == "strptime") {
+    using Options = arrow::compute::StrptimeOptions;
```

Comment:
Does `StrptimeOptions` have a `Defaults()` method in C++? If so, we should call it.

Reply:
Do I understand correctly that in `scalar_string.cc` line 1744 would suggest `StrptimeOptions` do not have `Defaults()` in C++?

Reply:
Yeah looks like it. I made ARROW-12809 to evaluate whether that's correct, but for the purposes of the PR, it's fine.